tests/basic_relay.go

```diff
@@ -0,0 +1,298 @@
+		relayerCancel       context.CancelFunc
+	)
+
+	subnetAInfo := teleporterTestUtils.GetSubnetATestInfo()
+	subnetBInfo := teleporterTestUtils.GetSubnetBTestInfo()
+	teleporterContractAddress := teleporterTestUtils.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := teleporterTestUtils.GetFundedAccountInfo()
+
+	teleporterMessage := teleporter.TeleporterMessage{
+		MessageID:               big.NewInt(1),
+		SenderAddress:           fundedAddress,
+		DestinationAddress:      fundedAddress,
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Receipts:                []teleporter.TeleporterMessageReceipt{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	//
+	// Set up relayer config
+	//
+	hostA, portA, err := teleporterTestUtils.GetURIHostAndPort(subnetAInfo.ChainNodeURIs[0])
+	Expect(err).Should(BeNil())
+
+	hostB, portB, err := teleporterTestUtils.GetURIHostAndPort(subnetBInfo.ChainNodeURIs[0])
+	Expect(err).Should(BeNil())
+
+	log.Info(
+		"Setting up relayer config",
+		"hostA", hostA,
+		"portA", portA,
+		"blockChainA", subnetAInfo.BlockchainID.String(),
+		"hostB", hostB,
+		"portB", portB,
+		"blockChainB", subnetBInfo.BlockchainID.String(),
+		"subnetA", subnetAInfo.SubnetID.String(),
+		"subnetB", subnetBInfo.SubnetID.String(),
+	)
+
+	relayerConfig := config.Config{
+		LogLevel:          logging.Info.LowerString(),
+		NetworkID:         peers.LocalNetworkID,
+		PChainAPIURL:      subnetAInfo.ChainNodeURIs[0],
+		EncryptConnection: false,
+		StorageLocation:   storageLocation,
+		SourceSubnets: []config.SourceSubnet{
+			{
+				SubnetID:          subnetAInfo.SubnetID.String(),
+				ChainID:           subnetAInfo.BlockchainID.String(),
+				VM:                config.EVM.String(),
+				EncryptConnection: false,
+				APINodeHost:       hostA,
+				APINodePort:       portA,
+				MessageContracts: map[string]config.MessageProtocolConfig{
+					teleporterContractAddress.Hex(): {
+						MessageFormat: config.TELEPORTER.String(),
+						Settings: map[string]interface{}{
+							"reward-address": fundedAddress.Hex(),
+						},
+					},
+				},
+			},
+		},
+		DestinationSubnets: []config.DestinationSubnet{
+			{
+				SubnetID:          subnetBInfo.SubnetID.String(),
+				ChainID:           subnetBInfo.BlockchainID.String(),
+				VM:                config.EVM.String(),
+				EncryptConnection: false,
+				APINodeHost:       hostB,
+				APINodePort:       portB,
+				AccountPrivateKey: hex.EncodeToString(fundedKey.D.Bytes()),
+			},
+		},
+	}
+
+	data, err := json.MarshalIndent(relayerConfig, "", "\t")
+	Expect(err).Should(BeNil())
+
+	f, err := os.CreateTemp(os.TempDir(), "relayer-config.json")
+	Expect(err).Should(BeNil())
+
+	_, err = f.Write(data)
+	Expect(err).Should(BeNil())
+	relayerConfigPath := f.Name()
+
+	log.Info("Created awm-relayer config", "configPath", relayerConfigPath, "config", string(data))
+
+	//
+	// Build Relayer
+	//
+	// Build the awm-relayer binary
```

Comment:
I think we can include building the binary in utils `RunRelayerExecutable`

Reply:
We run the relayer executable multiple times in the test in order to reset the configuration, but we don't rebuild in between. I think we should keep the build and run steps separate.

Reply:
nvm see that it's used separately later for checking already delivered msg